docs/markdown/Writing Plugins/tutorials/01-create-new-goal-with-a-custom-target.md

```diff
@@ -0,0 +1,335 @@
+
+```python
+@goal_rule
+async def goal_show_project_version(console: Console, targets: Targets) -> ProjectVersionGoal:
+    targets = [tgt for tgt in targets if tgt.alias == ProjectVersionTarget.alias]
+    for target in targets:
+        project_version = await Get(ProjectVersionFileView, ProjectVersionTarget, target)
+        console.print_stdout(project_version)
+    return ProjectVersionGoal(exit_code=0)
+```
+
+Understanding that calling `Get()` is what causes a particular `@rule` to be executed is essential! It may feel awkward that you cannot run your function. However, by using the `Get()`, you are asking Pants to run your rule, and only knowing this will get you quite far! 
+
+Compare this `Get()` call with the rule signature:
+
+```python
+# requesting an object of type "ProjectVersionFileView", passing an object of type "ProjectVersionTarget" in the variable "target"
+Get(ProjectVersionFileView, ProjectVersionTarget, target)
+```
+
+```python
+# it requires an object of type "ProjectVersionTarget" and will return an object of type "ProjectVersionFileView"
+@rule
+async def get_project_version_file_view(target: ProjectVersionTarget) -> ProjectVersionFileView: ...
+```
+
+The `project-version` Pants goal now shows some useful information -- the target path along with a dummy version. This means our `@rule` was run!
+
+```
+$ ./pants project-version myapp
+ProjectVersionFileView(path='myapp:main-project-version', version='1.2.3')
+```
+
+## Reading the `VERSION` file
+
+Let's read the `VERSION` file and print the version number in the terminal. The `source` field of our target needs to be ["hydrated"](https://www.pantsbuild.org/docs/rules-api-and-target-api#sourcesfield). [Reading a file](https://www.pantsbuild.org/docs/rules-api-file-system) is pretty straightforward as well. We use `Get()` to transform our inputs as needed. Knowing what class you need to request may be tricky, so make sure to review the documentation, and ask for help if you are stuck!    
+
+```python
+@rule
+async def get_project_version_file_view(
+    target: ProjectVersionTarget,
+) -> ProjectVersionFileView:
+    sources = await Get(HydratedSources, HydrateSourcesRequest(target[SourcesField]))
+    digest_contents = await Get(DigestContents, Digest, sources.snapshot.digest)
+    if len(digest_contents) > 1:
+        raise ValueError("There should be only one source file.")
+    file_content = digest_contents[0]    
+    return ProjectVersionFileView(
+        path=file_content.path, version=file_content.content.decode("utf-8").strip()
+    )
+```
+
+You would normally expect for a project to have only a single `version_file` target declared, so you could raise an exception if there are multiple targets of this type found within a single project. We could also replace our `Get()` calls in the `for` loop with a [`MultiGet()` call](https://www.pantsbuild.org/docs/rules-api-tips#tip-use-multiget-for-increased-concurrency):
```

Comment:
The transition from "maybe we only support one target" to "we could use `MultiGet`" makes all sorts of weird connections for me.

I think it is most valuable to introduce the notion of MultiGet here, I'm just pointing out that it may be confusing as currently presented here.

Reply:
Super helpful, indeed, that was weird. I've split those sections. Fixed!